platform/engineering/design-docs/2021-08-01-autonomous-deployments.md

```diff
@@ -0,0 +1,220 @@
+
+## Specifics
+
+### Detailed Design
+
+**Steps for a production deployment:**
+
+VSP maintains the ability to create and deploy full production builds as a failsafe until the system is stable.
+
+1. Merge feature branch into master to trigger workflow
+2. Is the daily deployment running and/or is master running on this commit?
+  - **Yes**: Pause until finished
+  - **No**: Continue
+3. Has an approved application changed?
+  - **Yes**: Continue
+  - **No**: Stop
+4. Build changed applications / Run unit tests
+5. Sync CSS/JS/images to prod S3 bucket
+
+Manual deployments follow the same process, but do not need to check the commit status. Should be able to trigger via the Github action UI and provide an app name (or list of apps). Teams should be made aware that if they manually deploy something that's running in master CI, the manual deployment results will be overwritten when the auto-deploy completes.
+
+**Steps for a staging/dev deployment:**
+
+VSP maintains the ability to create and deploy full staging/dev builds as a failsafe.
+
+1. Merge branch into staging/dev (locally or through PR)
+2. Build all applications
+3. If a deployment to staging/dev in in progress wait in a queue
+4. Sync /build/vagov{env} to staging/dev S3 bucket
+
+This will likely cause `staging` and `dev` to diverge from `master` over time. We can set up a scheduled job to merge master into staging/dev on a weekly basis and reserve the ability to completely reset the environments if needed. When an environment is reset, a slack notification should go out alerting VFS teams that they will need to remerge any branches. An alert should also go out to Slack the previous day if any preparations are needed as to not interfere with potential planned demos. 
+
+We will also require commits to be merged into `staging` (not `dev`) before they are allowed to merge into `master`. After this requirement is in place, we would like to monitor for feedback to see if we need to allow teams to bypass this requirement with branch naming `hotfix/` and/or `revert/` prefixes.
+
+#### Roadmap
+
+1. Update current system to function with new workflows
+2. Create and release staging/dev deployment behaviors
+3. Create and release initial production behaviors with "allow" list
+4. Create and release MVP production behaviors with "disallow" list
+
+#### Technologies Being Used
+
+- [Github Actions](https://github.com/features/actions)
+- [AWS](https://docs.aws.amazon.com/AmazonS3/latest/userguide/Welcome.html)
+- [Webpack](https://webpack.js.org/)
+
+### Work Estimates
+
```

Comment:
I would take the Work Estimates section out of this document. It doesn't belong in a design doc. If you'd like to, migrate it  to a separate document.